metaflow/plugins/cards/card_datastore.py

```diff
@@ -171,17 +170,19 @@ def _list_card_paths(self, card_type=None, card_hash=None):
             if card_type is not None and card_info.type != card_type:
                 continue
             elif card_hash is not None:
-                if (
-                    card_info.hash != card_hash
-                    and card_hash != card_info.hash[:NUM_SHORT_HASH_CHARS]
-                ):
+                if not card_info.hash.startswith(card_hash):
                     continue
+            elif card_id is not None and card_info.id != card_id:
+                continue
 
             if task_card_path.is_file:
                 cards_found.append(card_path)
 
         return cards_found
 
+    def create_full_path(self, card_path):
```

Comment:
This may be a stupid comment so ignore it if it is but I thought we had a separate value for card root (`METAFLOW_CARD_S3ROOT`). Granted, it defaults to the one for the datastore but shouldn't we use the actual value in case it isn't like the one for the datastore?

Reply:
Given your reliance on the flow_datastore, it may be better to not leave this as an option to users (and maybe only allow them to configure the card sub-directory -- with restrictions that it should probably not be a valid python name to avoid conflicts with a flow).

Reply:
I actually set the storage root [here](https://github.com/Netflix/metaflow/blob/b38c9aa25ef1a189dd9ba7a993533f1fafeaf4eb/metaflow/plugins/cards/card_cli.py#L340). That's why the path generally lands up correct even when `METAFLOW_CARD_S3ROOT` is set.